### PR TITLE
move openssl calls to common/libssl.c, check for defines

### PR DESCRIPTION
**Pull request by jsorg71**

### CODE REVIEW

**Comment by speidy** [comment]
+1 :)

**Comment by metalefty** [comment]
I haven't tested the change actually but looks good. Thank you for cleaning up my code!

File: common/ssl_calls.c [hunk 1]
```diff
@@ -843,3 +843,82 @@ ssl_get_cipher_name(const struct ssl_st *ssl)
...
+    protocols |= SSL_OP_NO_TLSv1_2;
+#endif
+    bad_protocols = protocols;
+    if (g_pos(str, ",TLSv1.2,") >= 0)
+    {
+#if defined(SSL_OP_NO_TLSv1_2)
+        log_message(LOG_LEVEL_DEBUG, "TLSv1.2 enabled");
+        protocols &= ~SSL_OP_NO_TLSv1_2;
+#else
+        log_message(LOG_LEVEL_DEBUG, "TLSv1.2 not enabled, not available");
```
Comment (metalefty) [inline common/ssl_calls.c]: I think something like "TLSv1.2 enabled by config, but not supported by system OpenSSL" will describe the situation better.

**Comment by metalefty** [comment]
I'm OK with this except for log messages. Merge?

**Comment by metalefty** [comment]
I'm going to make changes to ssl_calls. Not to cause conflicts, let me merge.

**Pull request merged**